src/transformers/generation/utils.py

```diff
@@ -1803,17 +1803,26 @@ def generate(
                 f"Passing both `cache_implementation` (used to initialize certain caches) and `{cache_name}` (a "
                 "Cache object) is unsupported. Please use only one of the two."
             )
+        elif not generation_config.use_cache:
```

Comment:
This was addressed in https://github.com/huggingface/transformers/pull/32863 -- sorry, should have tagged you to review!

There will be a significant rebase to be done in this file, as these cache preparation operations were moved into a new function -- have a look at the PR 🤗

Reply:
oke, will rebase main